### DISCUSSION TITLE
BelongsToSelect on thousands of records

### DISCUSSION THREAD

**Discussion by oxytoxin** [filamentphp/filament#326] [General]
I have a situation where a BelongsToSelect references thousands of records. When I use the search functionality, there isn't any problem if I type fast enough that it would only return a few results. But if I just type one letter, and it returns a lot of results, the page becomes unresponsive. How would you go about this problem?

**Comment by gavinhewitt** [comment]
Not sure if this is the preferred method but it works for me. Just add a limit to the results:

```
Components\BelongsToSelect::make('item')
    ->relationship('related', 'title', function ($query) {
        return $query->take(30);
    }),
```

> I have a situation where a BelongsToSelect references thousands of records. When I use the search functionality, there isn't any problem if I type fast enough that it would only return a few results. But if I just type one letter, and it returns a lot of results, the page becomes unresponsive. How would you go about this problem?

**Reply by oxytoxin** [reply]
Oh wow, I think I'm cool with this one. Thanks a lot man.
